src/io_threads.c

```diff
@@ -345,6 +345,7 @@ int trySendReadToIOThreads(client *c) {
     c->cur_tid = tid;
     c->read_flags = canParseCommand(c) ? 0 : READ_FLAGS_DONT_PARSE;
     c->read_flags |= authRequired(c) ? READ_FLAGS_AUTH_REQUIRED : 0;
+    c->read_flags |= c->flag.primary ? READ_FLAGS_PRIMARY : 0;
```

Comment:
This and the replica flag aren't really read or write flags. Maybe we should just have these act as a separate set of flags used by the IO system that are just set once?

Reply:
It would require an additional flag that needs to be turned on/off everywhere we toggle between primary and replica. I'm not entirely sure this would be the better approach.